tests/functional/test_new_resolver.py

```diff
@@ -1132,3 +1132,91 @@ def test_new_resolver_check_wheel_version_normalized(
         "simple"
     )
     assert_installed(script, simple="0.1.0+local.1")
+
+
+def test_new_resolver_does_not_reinstall_local_wheels(script):
```

Comment:
As I said before, I don't actually think we should have a test for this at all - we shouldn't be making guarantees about whether we do or don't reinstall. But having said that, I'm happy with adding this so that if we ever do accidentally reinstall, we'll catch it (we can remove the test if it was a deliberate decision - as long as we don't fall into the trap of "we test for it so it must be a guarantee" 🙁).

Reply:
LOL. I'm happy to remove tests. :)